test/e2e/framework/node/wait.go

```diff
@@ -206,3 +206,22 @@ func checkWaitListSchedulableNodes(c clientset.Interface) (*v1.NodeList, error)
 	}
 	return nodes, nil
 }
+
+// ReadyForTests determines whether or not we should continue waiting for the nodes
+// to enter a testable state. By default this means it is schedulable, NodeReady, and untainted.
+// Nodes with taints matching the whitelistedTaintRegexp are permitted to have that taint and
+// also have their node.Spec.Unschedulable field ignored for the purposes of this function.
+func ReadyForTests(node *v1.Node, whitelistedTaints *regexp.Regexp) bool {
+	if hasWhitelistedTaint(node, whitelistedTaints) {
+		// If the node has one of the whitelisted taints; just check that it is ready
+		// and don't require node.Spec.Unschedulable to be set either way.
+		if !IsNodeReady(node) || !IsNodeUntaintedWhitelist(node, whitelistedTaints) {
+			return false
+		}
+	} else {
+		if !IsNodeSchedulable(node) || !IsNodeUntainted(node) {
```

Comment:
What happens with the master tainted nodes?

Reply:
I'm not sure I understand the question.

If you have a node that is tainted as `node-role.kubernetes.io/master`, that is also the default for the whitelisted taint regexp. This would mean that we would use the path above (not `else` path this comment is on).

The end result would be that we would just check that it is ready and doesn't have other NoSchedule taints on it besides that one. Since it doesnt, this method would return `true` and the node would not be considered a blocker for the tests starting up.

This will become more clear when I extract the multi-node logic into a more testable situation. That way we can see at a high level when the cluster would start up vs keep polling.